src/util.ts

```diff
@@ -510,7 +512,7 @@ export class Util {
      * |  bbb | bb   |  bbb |  b   | bbb  |    bb |  bb   |     b | a     |
      * ```
      */
-    public rangesIntersect(a: Range, b: Range) {
+    public rangesIntersect(a: Range | undefined, b: Range | undefined) {
```

Comment:
This and `rangesIntersectOrTouch` are called with potentially `undefined` values elsewhere but handle undefinedness gracefully internally, so I've loosened their type signatures.